include/wex/presets/GameObject.hpp

```diff
@@ -0,0 +1,42 @@
+#include "../Common.hpp"
+#include "../Engine.hpp"
+#include "../util.hpp"
+#include "Component.hpp"
+#include <cstddef>
+#include <vector>
+
+namespace wex {
+
+/// \brief The game game object class. A game object is a bag of components with an
+/// `onUpdate` method that is called once per iteration of the game loop.
+class GameObject {
+	std::vector<std::unique_ptr<Component>> mComponents;
+
+  public:
+	GameObject() = default;
+
+	/// \brief called on every 'update' call to the parent engine
+	virtual void onUpdate([[maybe_unused]] double dt) {}
+
+	template <typename T>
+	T* get() {
+		static_assert(std::is_base_of_v<Component, T>);
+		/// TODO: optimize this O(n) lookup
+		for (auto const& compPtr : mComponents) {
+			auto comp = dynamic_cast<T*>(compPtr.get());
+			if (comp != nullptr) return reinterpret_cast<T*>(comp);
```

Comment:
Pls no `reinterpret_cast` 😭
Here you can just `return comp`

Reply:
lol, alright. Why is reinterpret_case bad?

Reply:
It's basically `(foo)bar`: there are no checks whatsoever, and unlike in C, materializing an object (or a pointer to one) that was never created before is UB in C++.